appium/webdriver/extensions/flutter/flutter_commands.py

```diff
@@ -0,0 +1,166 @@
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+from typing import Any, Optional, Tuple, Union
+from appium.webdriver.flutter_finder import FlutterFinder
+from appium.webdriver.webdriver import WebDriver
+from appium.webdriver.webelement import WebElement
+
+
+class FlutterCommand():
+    
+    def __init__(self, driver: WebDriver) -> None:
+        self.driver = driver
+        
+     # wait commands   
+     
+    def wait_for_visible(self, locator: Union[WebElement, FlutterFinder], time_out: Optional[int] = None) -> None:
+        """
+        Waits for a element to become visible.
+
+        Args:
+            locator: The element to wait for; can be a WebElement or a FlutterFinder.
+            time_out: Maximum wait time in seconds. Defaults to a predefined timeout if not specified.
+            
+        Returns: 
+            None: 
+        """
+        if isinstance(locator, WebElement):
+            opts = {'element': locator, 'timeout': time_out}
+        else:
+            opts = {'locator': locator.to_dict(), 'timeout': time_out}
+
+        self.execute_flutter_command('waitForVisible', opts)
+
+            
+    def wait_for_invisible(self, locator: Union[WebElement, FlutterFinder], time_out: Optional[int] = None) -> None:
+        """
+        Waits for a element to become invisible.
+
+        Args:
+            locator: The element to wait for; can be a WebElement or a FlutterFinder.
+            time_out: Maximum wait time in seconds. Defaults to a predefined timeout if not specified.
+            
+        Returns:
+            None: 
+        """
+        if isinstance(locator, WebElement):
+            opts = {'element': locator, 'timeout': time_out}
+        else:
+            opts = {'locator': locator.to_dict(), 'timeout': time_out}
+        
+        self.execute_flutter_command('waitForAbsent', opts)
+
+    # flutter action commands
+    
+    def perform_double_click(self, element: WebElement, offset: Optional[Tuple[int, int]] = None) -> None:
+        """
+        Performs a double-click on the given element, with an optional offset.
+
+        Args:
+            element: The element to double-click on. This parameter is required.
+            offset: The x and y offsets from the element to click at. If not specified, the click is performed at the element's center.
+            
+        Returns:
+            None:
+        """
+        opts = {'origin': element}
+        if offset:
```

Comment:
prefer to use explicit `is not None` comparison

Reply:
Thanks! Updated